ShippingApp/src/test/java/com/javaschool/shipping/ShippingAppApplicationTests.java

```diff
@@ -0,0 +1,63 @@
+package com.javaschool.shipping;
+
+import com.javaschool.common.QueueException;
+import com.javaschool.queue.QueueSender;
+import com.javaschool.service.ShippingService;
+import com.javaschool.service.ShippingServiceImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ShippingAppApplicationTests {
+
+    private QueueSender queueSender;
+    private AmqpTemplate amqpTemplateMock;
+    private ShippingService shippingService;
+
+    @Before
+    public void setup() {
+        amqpTemplateMock = mock(RabbitTemplate.class);
+        queueSender = new QueueSender(amqpTemplateMock);
+        shippingService = new ShippingServiceImpl(queueSender);
+    }
+
+    @Test
+    public void getSizeTest() {
+        String queueResponse = "[{\"id\":1,\"description\":\"Small\",\"priceFactor\":5},{\"id\":2,\"description\":\"Medium\",\"priceFactor\":10},{\"id\":3,\"description\":\"Large\",\"priceFactor\":15}]";
+        String queueMessage = "{\"type\":\"packageSize\"}";
+        when(amqpTemplateMock.convertSendAndReceive(null, null, queueMessage)).thenReturn(queueResponse);
+        List<String> sizes = shippingService.getSize();
+        assertEquals(sizes.size(), 3);
+    }
+
+    @Test(expected = QueueException.class)
+    public void getSizeExceptionTest(){
+        String queueMessage = "{\"type\":\"packageSize\"}";
+        when(amqpTemplateMock.convertSendAndReceive(null, null, queueMessage)).thenReturn(null);
+        List<String> sizes = shippingService.getSize();
+    }
+
+    @Test
+    public void getTypeTest() {
+        String queueResponse = "[{\"id\":2,\"description\":\"Box\",\"price\":10},{\"id\":3,\"description\":\"Envelope\",\"price\":5}]";
+        String queueMessage = "{\"type\":\"packageType\"}";
+        when(amqpTemplateMock.convertSendAndReceive(null, null, queueMessage)).thenReturn(queueResponse);
+        List<String> types = shippingService.getType();
+        assertEquals(types.size(), 2);
```

Comment:
Also assert the types are the ones you declared in the JSON

Reply:
The test is added to verify the items in the list, the unit test for the type that returns the method is not added because it is already specified in the method and the test is considered unnecessary.